### PR TITLE
1545 mw ip blacklisting

### PR DESCRIPTION
**Pull request by asoorm**
refs: #1545
    
This PR adds IP Blacklisting Middleware. It works in similar way to
IP Whitelisting.

Once merged:

- requires dashboard update to API Definition
- Requires UI
- Requires documentation

### CODE REVIEW

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -376,6 +376,8 @@ type APIDefinition struct {
 	EnableBatchRequestSupport bool                   `bson:"enable_batch_request_support" json:"enable_batch_request_support"`
 	EnableIpWhiteListing      bool                   `mapstructure:"enable_ip_whitelisting" bson:"enable_ip_whitelisting" json:"enable_ip_whitelisting"`
 	AllowedIPs                []string               `mapstructure:"allowed_ips" bson:"allowed_ips" json:"allowed_ips"`
+	EnableIpBlacklisting      bool                   `mapstructure:"enable_ip_blacklisting" bson:"enable_ip_blacklisting" json:"enable_ip_blacklisting"`
```
Comment (matiasinsaurralde) [inline apidef/api_definitions.go]: Perhaps `EnableIPBlacklisting`?

Comment (asoorm) [reply apidef/api_definitions.go]: I was wondering the exact same thing and after checking, the word *blacklist* is both a noun and a verb. https://en.oxforddictionaries.com/definition/blacklist .

Comment (asoorm) [reply apidef/api_definitions.go]: I kept in same format as `EnableIpWhiteListing`

**Comment by buger** [comment]
Looks good!

**Pull request merged**